src/url.js

```diff
@@ -321,6 +321,19 @@ export function isLocalhostOrigin(url) {
 }
 
 /**
+ * Returns whether the URL has valid protocol.
+ * @param {string|!Location} url URL of an AMP document.
+ * @return {boolean}
+ */
+export function isProtocolValid(url) {
```

Comment:
please add bunch of tests for this in a separate PR today (let's merge this PR to cherry pick)